golem-worker-executor-base/src/services/scheduler.rs

```diff
@@ -68,16 +68,22 @@ impl SchedulerServiceDefault {
             worker_activator,
         };
         let svc = Arc::new(svc);
-        let svc_clone = svc.clone();
-        let background_handle = tokio::spawn(async move {
-            loop {
-                tokio::time::sleep(process_interval).await;
-                let r = svc_clone.process(Utc::now()).await;
-                if let Err(err) = r {
-                    error!("Error in scheduler background task: {}", err);
+        let background_handle = {
+            let svc = svc.clone();
+            tokio::spawn(async move {
+                loop {
+                    tokio::time::sleep(process_interval).await;
+                    if svc.shard_service.is_ready() {
+                        let r = svc.process(Utc::now()).await;
+                        if let Err(err) = r {
+                            error!(err, "Error in scheduler background task");
+                        }
+                    } else {
+                        warn!("Skipping schedule, shard service is not ready")
```

Comment:
This will produce a lot of warnings in "normal" scaling up/down situations I think, but we can leave it for now